bot/events/users/play.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import typing as t
+
+from nextcord import Message
+from nextcord.ext.commands import Bot
+
+from bot.config import Config
+from bot.errors import filter
+from bot.src.yt_link_changer import YTLinks
+
+
+log = logging.getLogger(__name__)
+
+
+async def get_link(message: Message, links: t.List[str], titles: t.List[str]) -> t.Optional[str]:
+    num = int(message.content)
+    if num in range(1, Config.result_len + 1):
+        await message.channel.send(f'You choose {num} - {titles[num-1]}:{links[num-1]}')
+        return links[num-1]
+    await message.channel.send(
+        f'Abort! :anger: You choose {num} - which is not valid,'
+        f'type number only in range(1-{Config.result_len}) :anger:'
+    )
+
+
+async def link_extractor(bot: Bot, message: Message, request_proc: YTLinks, text: str) -> t.Optional[str]:
+    titles, links = request_proc.yt_search(text, Config.result_len)
+    choice_dialog = f'Type num of the song 1-{Config.result_len} (example: 2) \n' + ' '.join(titles)
+    await message.channel.send(choice_dialog)
+
+    def is_my_message(msg):
+        return msg.author == message.author and msg.channel == message.channel
+    user_message = await bot.wait_for('message', check=is_my_message)
+
+    try:
+        return await get_link(user_message, links, titles)
+    except ValueError:
+        await message.channel.send(
+            f'{user_message.content} mean that hoisting crane, but you are adopted :scream_cat: ))'
+        )
+
+
+def play(bot: Bot):
+    async def on_message(message: Message):
+        if message.content.startswith('?play'):
+            command, *text = message.content.split()
+            text = ' '.join(text)
+            request_proc = YTLinks()
+            try:
+                result = request_proc.filter(text)
+            except filter.FilterStrLenError:
+                await message.channel.send(f'Query lenght is too long(more than {Config.query_len} symbols)')
+                return
+            if not result:
+                result = await link_extractor(bot, message, request_proc, text)
```

Comment:
if you use request_proc in this func maybe implement this in YTLinks?

Reply:
Unga not bunga
![image](https://user-images.githubusercontent.com/71280626/195418775-18f8ddd4-9d52-4441-a387-ae2b6526503a.png)